torch/fx/experimental/symbolic_shapes.py

```diff
@@ -111,10 +111,22 @@ class PySymInt(object):
     our program. They're what sit under FakeTensor, and contains our primary
     implementation of symbolic shapes.
     """
-    def __init__(self, expr, shape_env, constant=None):
+    def __init__(self, expr, shape_env, constant=None, ref_id=None, kind=None, idx=None):
         self._expr = expr
         self.shape_env = shape_env
         self.constant = constant
+        self.ref_id = ref_id
+        self.kind = kind
+        self.idx = idx
+        if self.ref_id and str(self.expr) not in ('False', 'True'):
+            if self.expr not in self.shape_env.expr_to_id:
+                self.shape_env.expr_to_id[self.expr] = set()
+            self.shape_env.expr_to_id[self.expr].add((self.ref_id, self.kind, self.idx))
+
+            if self.ref_id not in self.shape_env.expr_to_id:
+                self.shape_env.expr_to_id[self.ref_id] = set()
+            self.shape_env.expr_to_id[self.ref_id].add((self.expr, self.kind, self.idx))
```

Comment:
I don't really understand what these maps are for

Reply:
See updated doc.